core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/action/ImportDictionaryServlet.java

```diff
@@ -0,0 +1,132 @@
+        String path = request.getParameter("dictionary");
+        RequestParameter csvfile = request.getRequestParameter("./csvfile");
+
+        if (csvfile != null) {
+            InputStream csvContent = csvfile.getInputStream();
+
+            List<String> languages = new ArrayList<>();
+            List<String> KEYs = new ArrayList<>();
+            List<List<String>> translations = new ArrayList<>();
+
+            ResourceResolver resourceResolver = request.getResourceResolver();
+
+            try (InputStreamReader reader = new InputStreamReader(csvContent)) {
+                String result = IOUtils.toString(csvContent, String.valueOf(StandardCharsets.UTF_8));
+                csvContent.reset();
+                CSVFormat format = null;
+                if (result.contains(";")) {
+                    format = CSVFormat.newFormat(';').withFirstRecordAsHeader();
+                } else if (result.contains(",")){
+                    format = CSVFormat.DEFAULT.withFirstRecordAsHeader();
+                } else {
+                    String error = "Invalid CSV file. The Delimiter should be ',' or ';'.";
+                    LOG.error(error);
+                    response.setStatus(400, error);
+                    return;
+                }
+                CSVParser csvParser = new CSVParser(reader, format);
+                Map<String, Integer> headers = csvParser.getHeaderMap();
+
+                if (!headers.containsKey("KEY") || headers.get("KEY") != 0) {
+                    String error = "Invalid CSV file. The first column must be 'KEY'. The Delimiter should be ',' or ';'.";
+                    LOG.error(error);
+                    response.setStatus(400, error);
+                    return;
+                }
+
+                headers.remove("KEY");
+                for (String language : headers.keySet()) {
+                    languages.add(language);
+                    translations.add(new ArrayList<>());
+                }
+
+                for (CSVRecord record : csvParser) {
+                    if (record.size() != languages.size() + 1) {  // +1 for the 'KEY' column.
+                        LOG.warn("Ignoring row with incorrect number of translations: " + record);
+                        continue;
+                    }
+
+                    String label = record.get("KEY");
+                    KEYs.add(label);
+
+                    for (String language : languages) {
+                        int index = languages.indexOf(language);
+                        String translation = record.get(language);
+                        translations.get(index).add(translation);
+
+                        Resource languageResource = resourceResolver.getResource(path + "/" + language);
+                        if (languageResource == null) {
+                            Map<String, Object> newFolderProperties = new HashMap<>();
```

Comment:
no need for new prefix

Reply:
removed completely with the use of a Map.of() instead